### PR TITLE
api plugin arrange

### PR DESCRIPTION
**Pull request by upccup**

### CODE REVIEW

**Comment by vitan** [comment]
LGTM

but we need another PR to improve the params of the plugin/Init() function:

Init(Param_1, Param_2, Param_3)

File: src/api/router.go [hunk 1]
```diff
@@ -126,6 +85,28 @@ func (api *Api) ApiRouter() *gin.Engine {
 		v1.GET("/stacks/:namespace/services/:service_id/cd_url", api.ServiceCDAddr)
 	}
 
+	for _, plugin := range apiplugin.ApiPlugins {
+		if plugin.Instance != nil {
+			switch plugin.Name {
+			case apiplugin.Search:
+				searchApi, ok := plugin.Instance.(*search.SearchApi)
```
Comment (cmingxu) [inline src/api/router.go]: How about add this section as a callback of apiplugin Init?

such as apiplugin.InitCallback = func() {
...
}

then call a apiplugin.InitCallback here

**Comment by cmingxu** [comment]
Still we don't have a good way of separation  between a plugin and main program, somehow mixed together both design and coding perspective.

File: src/api/router.go (hunk 1 above)
Comment (upccup) [inline src/api/router.go]: i think this is the plugin register not callback. ant we should add the plugins info to misc

**Comment by upccup** [comment]
@cmingxu   in my opinion that was we definition plugin too arbitrary. some plugin like a pkg such as registryauth, and we import it in the main program

**Pull request merged**